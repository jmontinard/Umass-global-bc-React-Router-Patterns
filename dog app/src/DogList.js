// DogList.js
import React from 'react';
import { Link } from 'react-router-dom';


const DogList = ({ dogs }) => {
  return (
    <div className="dog-list">
      <h1>Dog List</h1>
      <ul>
        {dogs.map(dog => (
          <li key={dog.name}>
            <Link to={`/dogs/${dog.name}`}>
              <img src={dog.src} alt={dog.name} />
              <p>{dog.name}</p>
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default DogList;

