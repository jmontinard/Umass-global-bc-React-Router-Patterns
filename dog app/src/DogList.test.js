import React from 'react';
import { render } from '@testing-library/react';
import DogList from './DogList';

test('renders DogList component without crashing', () => {
  render(<DogList dogs={[]} />);
});

test('matches snapshot for DogList component', () => {
  const { container } = render(<DogList dogs={[]} />);
  expect(container).toMatchSnapshot();
});
