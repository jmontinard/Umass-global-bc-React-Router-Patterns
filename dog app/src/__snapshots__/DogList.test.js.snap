// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for DogList component 1`] = `
<div>
  <div
    class="dog-list"
  >
    <h1>
      Dog List
    </h1>
    <ul />
  </div>
</div>
`;
